extensions/ql-vscode/src/query-history.ts

```diff
@@ -77,6 +86,9 @@ class HistoryTreeDataProvider
   constructor(private ctx: ExtensionContext) { }
 
   async getTreeItem(element: CompletedQuery): Promise<vscode.TreeItem> {
+    if (element.treeItem !== undefined)
```

Comment:
I think it would be nicer to convert `treeItem` to a getter on `CompletedQuery`. And move all this code into the getter. It avoids a type assertion on `updateTreeItemContextValue`.

Reply:
Good idea.

Reply:
Wait, no, this function needs to have the context of the treeview. The `CompletedQuery` doesn't have that information, or at least doesn't plausibly have it until the treeitem is created.